scripts/west_commands/runners/core.py

```diff
@@ -140,25 +142,41 @@ def get(self, option, *args):
         return self.options.get(option, *args)
 
     def _init(self):
-        self._parse(os.path.join(self.build_dir, 'zephyr', '.config'))
+        self._parse(self.path)
 
     def _parse(self, filename: str):
+        parser = re.compile("^(?P<prefix>#*)[ \t]*(?P<option>CONFIG_[a-zA-Z0-9_]+)=(?P<value>.*)")
```

Comment:
Thanks for the update!

I can't find anything in this PR that preserves the explicit match for `# CONFIG_FOO is not set` -> `CONFIG_FOO=n`, though; am I missing it?

Reply:
I have argued why the `is not set` match does not add value here https://github.com/zephyrproject-rtos/zephyr/pull/30214#discussion_r531233454, but If you really want it I can match the `is not set` lines to `None`.